mlflow/sklearn/__init__.py

```diff
@@ -908,14 +908,17 @@ def patched_fit(original, self, *args, **kwargs):
         "sklearn.feature_selection",
     ]
 
+    excluded_class_names = [
+        "sklearn.compose._column_transformer.ColumnTransformer",
+    ]
```

Comment:
`sklearn.compose` contains `TransformedTargetRegressor` which should be patched:

https://scikit-learn.org/stable/modules/generated/sklearn.compose.TransformedTargetRegressor.html#sklearn-compose-transformedtargetregressor